### ISSUE TITLE
Future plans

### ISSUE THREAD

**Issue by mattstauffer** [tighten/lambo#186]
I've *long* been considering building an all-in-one tool that sets up my ideal local Mac dev experience.

I'm considering Lambo for that.

It would:
- Bundle a few tools:
    - Valet
    - Laravel installer
    - Lambo's current functionality, I think
    - Takeout
- Allow users or packages to register custom commands (e.g. `lambo timezone EST` or `lambo pecl doThing`)

The problem is, I keep *thinking* about it, but I just don't know whether it's actually worth the effort. Are more people moving to Sail? Is there any need for a bundled set of tools? Would people use clever command line tooling more if they were just writing easy packages extending Lambo's already powerful tooling, or are people preferring GUIs more these days? Is this more targeted at newcomers, who are more using Sail? I have no idea.

**Comment by jonsugar** [comment]
@mattstauffer, we've mused over this a few times if I recall.

I'm thinking the same but also wrapping sail into the tool. Sail is still too much docker for me. Dare I say, it's not very Laravel? For me "being Laravel" means 1st class DX and the least friction possible. Valet, Takeout and Lambo are good examples of being "Laravel". The rails command for Ruby on Rails is another example of something that's "Laravel" (ignoring actual timelines and history of course).

Valet being mac-only frustrates me since I swap between Mac and Linux development but it has great DX. Sail works on Mac, Linux and Windows but its API is lacking in terms of DX. Takeout is just better for managing infrastructure dependencies and a joy to use. Lambo does some extra stuff the Laravel installer should™.

**TLDR, an ideal tool?**

- The project configurability of Lambo
- Ability to bootstrap a project with any/all of the first party Laravel packages
- Easily manage infrastructure dependencies like Takeout
- DX of Valet, Takeout and the Rails command
- Cross platform like Sail

There is a lot of work to get all that done. Is it worth the effort? That depends imho. Should Matt/Tighten do this, maybe not. Do I think Laravel/Taylor could? Maybe. The rails command is a gold standard of sorts I think and its maintained by them. 

**Disclaimer**: I am but one person on the internet and both Taylor and Matt are far more qualified to decide what tools are needed and what dev efforts are worth the ROI... these are just my thoughts on the issue.
